### DISCUSSION TITLE
Is there a simple way to generate multiple line segments?

### DISCUSSION THREAD

**Discussion by note286** [lvjr/tabularray#354] [Q&A]
Is there a simpler way to generate multiple line segments that repeats regularly like the following? Without having to write so many lines of almost repetitive code?

```latex
\documentclass{article}
\usepackage{tabularray}
\begin{document}
\begin{tblr}
{
colspec    = {cccccccc},
hline{1,Z} = {wd=.08em},
hline{2}   = {1-2}{wd=.05em,leftpos=-1,rightpos=-1,endpos=true},
hline{2}   = {3-4}{wd=.05em,leftpos=-1,rightpos=-1,endpos=true},
hline{2}   = {5-6}{wd=.05em,leftpos=-1,rightpos=-1,endpos=true},
hline{2}   = {7-8}{wd=.05em,leftpos=-1,rightpos=-1,endpos=true},
hline{2}   = {9-10}{wd=.05em,leftpos=-1,rightpos=-1,endpos=true}
}
1 & 2 & 3 & 4 & 5 & 6 & 7 & 8 & 9 & 10 \\
1 & 2 & 3 & 4 & 5 & 6 & 7 & 8 & 9 & 10 \\
\end{tblr}
\end{document}
```

![image](https://user-images.githubusercontent.com/16207292/214332553-a307feb2-7d91-449c-9e10-5d69a486f64e.png)

**Comment by lvjr** [comment]
This is related to issue #207.

**Reply by note286** [reply]
I looked at the contents in #207, but I couldn't find a solution. My demand is whether the following contents can be simplified. 

```latex
hline{2} = {1-2}{wd=.05em,leftpos=-1,rightpos=-1,endpos=true},
hline{2} = {3-4}{wd=.05em,leftpos=-1,rightpos=-1,endpos=true},
hline{2} = {5-6}{wd=.05em,leftpos=-1,rightpos=-1,endpos=true},
hline{2} = {7-8}{wd=.05em,leftpos=-1,rightpos=-1,endpos=true},
hline{2} = {9-10}{wd=.05em,leftpos=-1,rightpos=-1,endpos=true}
```

For example, reduced to

```latex
hline{2} = {1-2,3-4,5-6,7-8,9-10}{wd=.05em,leftpos=-1,rightpos=-1,endpos=true}
```

Of course, I know `{1-2,3-4,5-6,7-8,9-10}` not mean enumerating multiple identical styles, which will actually merge into one line, although this line may be interrupted. For example:

```latex
\documentclass{article}
\usepackage{tabularray}
\begin{document}
\begin{tblr}
{
colspec    = {cccccccc},
hline{1,Z} = {wd=.08em},
hline{2}   = {1-2,5-6,7-8,9-10}{wd=.05em,leftpos=-1,rightpos=-1,endpos=true}
}
1 & 2 & 3 & 4 & 5 & 6 & 7 & 8 & 9 & 10 \\
1 & 2 & 3 & 4 & 5 & 6 & 7 & 8 & 9 & 10 \\
\end{tblr}

\begin{tblr}
{
colspec    = {cccccccc},
hline{1,Z} = {wd=.08em},
hline{2}   = {1-2}{wd=.05em,leftpos=-1,rightpos=-1,endpos=true},
hline{2}   = {5-6}{wd=.05em,leftpos=-1,rightpos=-1,endpos=true},
hline{2}   = {7-8}{wd=.05em,leftpos=-1,rightpos=-1,endpos=true},
hline{2}   = {9-10}{wd=.05em,leftpos=-1,rightpos=-1,endpos=true}
}
1 & 2 & 3 & 4 & 5 & 6 & 7 & 8 & 9 & 10 \\
1 & 2 & 3 & 4 & 5 & 6 & 7 & 8 & 9 & 10 \\
\end{tblr}
\end{document}
```

![image](https://user-images.githubusercontent.com/16207292/214534435-d3967b41-89ec-4866-81c5-35a4a363f4c3.png)



`{1-2,5-6,7-8,9-10}` is a line only obmit the region of `3-4`.

In the above example, although `{1,Z}` and `{1-2,5-6,7-8,9-10}` are both comma-separated lists, their meanings are obviously different.

**Reply by lvjr** [reply]
You need to have a look at the linked [answer](https://tex.stackexchange.com/questions/634497/define-a-style-in-tabularray/634538#634538) in #207, and define your key-value `myhline = {2}{1-2,5-6,7-8,9-10}`.

**Reply by note286** [reply]
Thank you. Let me have a look.
